### PR TITLE
DB-11668 refactor SpliceAdmin / SpliceSystemProcedures

### PR DESCRIPTION
**Pull request by martinrupp**
## Description
Some simple refactoring:
`SpliceAdmin` contains nearly 3000 lines of code, with a lot of copied code.

`SpliceSystemProcedures` creates all the procedures even thought they might be better off in the repective classes (e.g. ReplicationSystemProcedure knows best about its own procedures, so it should add its functions to the procedure)

## How to test
(refactoring, no additional testing necessary)

### CODE REVIEW

File: splice_machine/src/main/java/com/splicemachine/derby/procedures/BaseAdminProcedures.java [hunk 1]
```diff
@@ -192,4 +192,56 @@ protected static ExecRow buildExecRow(ResultColumnDescriptor[] columns) throws S
...
+                        function.accept(hostAndPort, connection, rs);
+                    }
+                }
+            }
+        });
+    }
+
+    public static void execute(Connection connection, String sql) throws SQLException {
+        try (Statement stmt = connection.createStatement()) {
+            stmt.execute("call SYSCS_UTIL.INVALIDATE_DICTIONARY_CACHE()");
```
Comment (dgomezferro) [inline splice_machine/src/main/java/com/splicemachine/derby/procedures/BaseAdminProcedures.java:244]: Should this be `stmt.execute(sql);` ?

File: splice_machine/src/main/java/com/splicemachine/derby/procedures/BaseAdminProcedures.java [hunk 2]
```diff
@@ -192,4 +192,56 @@ protected static ExecRow buildExecRow(ResultColumnDescriptor[] columns) throws S
...
+            if(function == null)
+                execute(connection, sql);
+            else {
+                try (Statement stmt = connection.createStatement()) {
+
+                    try (ResultSet rs = stmt.executeQuery(sql)) {
+                        function.accept(hostAndPort, connection, rs);
+                    }
+                }
+            }
```
Comment (dgomezferro) [inline splice_machine/src/main/java/com/splicemachine/derby/procedures/BaseAdminProcedures.java:238]: Why not have two separate functions?

Comment (dgomezferro) [reply splice_machine/src/main/java/com/splicemachine/derby/procedures/BaseAdminProcedures.java:238]: It looks like function is never null, can you remove that branch?

File: splice_machine/src/main/java/com/splicemachine/derby/procedures/BaseAdminProcedures.java (hunk 1 above)
Comment (dgomezferro) [reply splice_machine/src/main/java/com/splicemachine/derby/procedures/BaseAdminProcedures.java:244]: If function is never null in the other call, this function is never used, so remove too?

**Review by dgomezferro — CHANGES_REQUESTED** [review]

File: splice_machine/src/main/java/com/splicemachine/derby/procedures/BaseAdminProcedures.java (hunk 1 above)
Comment (martinrupp) [reply splice_machine/src/main/java/com/splicemachine/derby/procedures/BaseAdminProcedures.java:244]: done (both)

File: splice_machine/src/main/java/com/splicemachine/derby/procedures/BaseAdminProcedures.java (hunk 2 above)
Comment (martinrupp) [reply splice_machine/src/main/java/com/splicemachine/derby/procedures/BaseAdminProcedures.java:238]: done

**Review by dgomezferro — APPROVED** [review]

**Review by ipraznik-splice — APPROVED** [review]

**Review by carolp-503 — APPROVED** [review]
+1

**Comment by martinrupp** [comment]
please also see ee branch https://github.com/splicemachine/spliceengine-ee/pull/637

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/2750/

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/2751/

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/2759/

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/2760/

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Pull request merged**